src/runtime/pprof/pprof.go

```diff
@@ -907,8 +907,7 @@ func writeProfileInternal(w io.Writer, debug int, name string, runtimeProfile fu
 }
 
 func scaleMutexProfile(cnt int64, ns float64) (int64, float64) {
-	period := runtime.SetMutexProfileFraction(-1)
-	return cnt * int64(period), ns * float64(period)
+	return cnt, ns
 }
```

Comment:
Both `scaleBlockProfile()` and `scaleMutexProfile()` are now no-ops like this which enables a simplifying refactoring. I.e. remove both functions as well as the `scaleProfile` arg from `writeProfileInternal()` and `printCountCycleProfile()`.

Note: I haven't tried this yet, just glanced over the code.